sympy/matrices/random.py

```diff
@@ -0,0 +1,2388 @@
+from ..core import I
+from ..core.mul import Mul
+from ..core.random import sample
+from ..core.sympify import _sympify as sympify
+from ..functions import sqrt, re, im, conjugate
+from ..tensor import shape
+from .dense import eye, zeros, diag
+
+__all__ = 'projection', 'jordan', 'transposition', \
+          'permutation', 'elementary', 'rotation', 'reflection', \
+          'diagonal_normal', 'jordan_normal', 'isometry_normal', \
+          'triangular', 'invertible', 'singular', \
+          'idempotent', 'nilpotent', 'diagonalizable', 'triangularizable', \
+          'orthogonal', 'unitary', 'normal', \
+          'symmetric', 'hermite', 'square', \
+          'regular_to_singular', 'complex_to_real'
+
+
+# === random number generator functions ===
+
+
+def _sample(scalars, k=None):
+    _k = k or 1
+    if hasattr(scalars, 'sample'):
+        smpl = scalars.sample(_k)
+    else:
+        smpl = sample(scalars, _k)
+    return smpl if k else smpl[0]
+
+
+# === helper ===
+
+
+_EPS = 2 * abs(abs(complex(1, 1) / abs(complex(1, 1))) - 1)
+"""numerical tolerance for float and complex types"""
+
+
+def _is_real(z):
+    """checks number to be real number, incl. complex without imaginary part"""
+    if isinstance(z, (int, float)):
+        return True
+    if isinstance(z, complex):
+        return not z.imag
+    z = sympify(z)
+    return not bool(z.is_complex) or bool(z.is_real)
+
+
+# === default sets ===
+
+
+_elementary_scalars = -1, 1
+_elementary_units = _elementary_scalars
+_rotation_scalars = (sqrt(2) / 2, sqrt(2) / 2), (0, -1),
+_unitary_scalars = tuple(c * 1 + s * I for c, s in _rotation_scalars)
+
+
+# === fundamental constructor ===
+
+
+def super_elementary(dim,
+                     index=None,
+                     value=None,
+                     *scalars):
```

Comment:
Every other function with a scalars argument takes a keyword argument rather than `*args`. It would seem more consistent for this one to do the same.

Reply:
no more relevant once `super_elementary` is removed